### DISCUSSION TITLE
Task inventory vmware

### DISCUSSION THREAD

**Discussion by sec1083** [glpi-project/glpi-agent#614] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

None

### Describe the bug

I am using glpi v10.0.11 and glpi inventory plugin v1.3.4. I created a task to inventory vmware, but the task does not execute. It is stoped in prepare mode. This task is the only task that does not work. The other tasks like netinventory, inventory and netdiscovery are running well.  

Available tasks :
- NetInventory (v6.1)
- WakeOnLan (v2.2)
- ESX (v2.10)
- Collect (v2.9)
- Inventory (v1.15)
- NetDiscovery (v6.1)
- RemoteInventory (v1.4)

I did not see nothing interesting in the logs about this error

### To reproduce

To reproduce you need to create a task on glpi to make an inventory from the vmware and execute it 

### Expected behavior

The expected behavior is the task running and somo informations on agent log file 

### Operating system

Linux

### GLPI Agent version

1.7

### GLPI version

10.0.11

### GLPIInventory plugin or FusionInventory for GLPI plugin version

GLPI Inventory v1.3.4

**Comment by sec1083** [comment]
I found [debug] Skipping fusioninventory-agent-task-esx package as not installed, status='install ok unpacked'  in the debbug log with the command glpi-agent --debug. But i tried using this task on fusioninventory agent but the result is the same on glpi inventory plugin.

**Comment by g-bougard** [comment]
Hi @sec1083 

you're providing no evidence of any bug in glpi-agent. I can only assume you did something wrong with the ESX task configuration in GLPI and if it's the case, you opened this issue at the wrong place.
Can you at least share the log part related to ESX task run ?

**Comment by sec1083** [comment]
![teste](https://github.com/glpi-project/glpi-agent/assets/160243417/5230d3d8-8be1-480a-9ec5-691aded4fda1)

Like i said, the unique task that does not working is this task. It stay in prepare mode more than days. The other tasks are working well, going from prepare to execute. Like i said i instaled all necessary packages and all is configured well. Despite all this, the task vmware still in mode prepare. You ask me about log, but i insist that the plugin is not working well with this specific task, the task simply do nothing. I think that if i try to execute in command line  glpi-esx --debug it will work well, but the plugin task exist to make this for me automatically

**Comment by sec1083** [comment]
I taked this log. Can somebody help me?

```
**[info] running task ESX**
[error] [http client] Can't decode JSON content, starting with:


<!DOCTYPE html>
<html lang="pt"       >
<head>
   <title>Autenticação - GLPI</title>

   <meta charset="utf-8" />
...
```
![image](https://github.com/glpi-project/glpi-agent/assets/160243417/6c850fbe-2007-440d-aff0-89e0074cce0a)

**Comment by g-bougard** [comment]
Hi @sec1083 

you're probably not using the right url but I can't be sure as you truncated the log. Check this documentation and read carefully the `Caution` note: https://glpi-agent.readthedocs.io/en/latest/configuration.html#server

**Comment by sec1083** [comment]
Hi @g-bougard 

I think if the url does not configured right all the tasks would not working, but the scenario is the opposite. All tasks are working well, less the vmware task.

**Comment by g-bougard** [comment]
Hi @sec1083 

all the tasks don't use the same protocol, so if you just think you're right and don't follow an expert advice for this reason, you will probably just miss an easy solution.

Actually, you didn't share a full log so we can't even tell you if the url is right or not. You only share a short part which shows us the agent is not reaching the server at the right place: so or the agent url is wrong, or the server configuration is wrong.

**Reply by sec1083** [reply]
I am not thinking i am right, i use glpi for years in production and i know how to configure and how it working. But i am testing the new version of glpi with glpi inventory plugin. In production i am using glpi 10.0.6 with fusioninventory and all tasks working well. The url is ok, is correct and working to communicate between agent and server. But i apreciate your help @sec1083

**Comment by JackTor** [comment]
Hi, i have the same problem

[root@infratest jatorres]#  glpi-agent --list-tasks

Available tasks :
- ESX (v2.10)
- Collect (v2.9)
- NetDiscovery (v6.1)
- RemoteInventory (v1.4)
- Inventory (v1.15)
- WakeOnLan (v2.2)
- Deploy (v3.0)
- NetInventory (v6.1)

target local0: local /tmp
Planned tasks: RemoteInventory,Inventory

target server0: server http://IP/glpi/marketplace/glpiinventory/
Planned tasks: ESX,Collect,NetDiscovery,RemoteInventory,Inventory,WakeOnLan,Deploy,NetInventory

glpi-agent version 1.7.3
glpi server 10.0.14
glpi plugin-inventory from marketplace: 1.3.5

I have created the task for de UI glpi server but this task not begin, stuck in **_prepared_**, when i execute for cli _glpi-agent --logger=stderr --debug --debug --tasks=ESX_  the output is:

[info] running task ESX
[debug2] [http client] Not using compression
[debug2] http://IP/glpi/marketplace/glpiinventory/?action=getConfig&task[ESX]=2.10&machineid=nameAgent-2023-09-28-14-36-13
[error] [http client] Can't decode JSON content, starting with:


<!DOCTYPE html>
<html lang="en"       >
<head>
   <title>Authentication - GLPI</title>

   <meta charset="utf-8" />
...

[info] ESX task not supported by server0

Why, what's wrong?   the URL is right according the doc https://glpi-agent.readthedocs.io/en/latest/configuration.html#server

If i execute the command manually for cli, it is Work:

glpi-esx --host vcenter --user xxx@vsphere.local  --password 'xxxx' --path /opt/esx/ --json --debug
glpi-injector -v -d  /opt/esx/ -u http://IP/glpi/marketplace/glpiinventory/

I need help!

**Reply by JackTor** [reply]
Hi everyone

I have found the solution!!

I had an error in my nginx configuration,  i maked the config as https://glpi-install.readthedocs.io/en/latest/prerequisites.html#nginx-configuration and change URL for the glpi-agent so, all task is working.

Before Nginx 
```
root /var/www/html/glpi;
location ~ \.php$ {
-------
}

```
After (Working)
```
root /var/www/html/glpi/public;
location ~ ^/index\.php$ {
-----
}

```
New URL fot the glpi-agents: http://ip/marketplace/glpiinventory/

Regards!!!!
